src/mpc-hc/DVBChannel.h

```diff
@@ -96,6 +96,39 @@ struct DVBStreamInfo {
     LCID GetLCID() { return ISO6392ToLcid(CStringA(sLanguage)); };
 };
 
+static std::vector<CString> Supported_Protocols = [] {
+    std::vector<CString> Protocols;
+    Protocols.push_back(_T("rtp://"));
+    Protocols.push_back(_T("udp://"));
+    Protocols.push_back(_T("rtsp://"));
+    Protocols.push_back(_T("mms://"));
+    Protocols.push_back(_T("http://"));
+    Protocols.push_back(_T("https://"));
+    Protocols.push_back(_T("rtspu://"));
+    Protocols.push_back(_T("rtspm://"));
+    Protocols.push_back(_T("rtspt://"));
+    Protocols.push_back(_T("rtsph://"));
+    Protocols.push_back(_T("RTP://"));
+    Protocols.push_back(_T("UDP://"));
+    Protocols.push_back(_T("RTSP://"));
+    Protocols.push_back(_T("MMS://"));
+    Protocols.push_back(_T("HTTP://"));
+    Protocols.push_back(_T("HTTPS://"));
+    Protocols.push_back(_T("RTSPU://"));
+    Protocols.push_back(_T("RTSPM://"));
+    Protocols.push_back(_T("RTSPT://"));
+    Protocols.push_back(_T("RTSPH://"));
+    return Protocols;
+} ();
+
+
+static bool IsValidUrl(LPCTSTR strUrl)
+{
+    auto it = find_if(Supported_Protocols.begin(), Supported_Protocols.end(),
+        [&](CString const & Protocol) { return (((CString)strUrl).Find(Protocol) == 0); });
+    return (it != Supported_Protocols.end());
+}
+

```

Comment:
If I may suggest something, I would use convert string to lower instead of listing all strings. Something like this should work. :) 

``` diff
diff --git a/src/mpc-hc/DVBChannel.h b/src/mpc-hc/DVBChannel.h
index de52ac6..d5055ed 100644
--- a/src/mpc-hc/DVBChannel.h
+++ b/src/mpc-hc/DVBChannel.h
@@ -21,6 +21,7 @@
 #pragma once
 
 #include <vector>
+#include "ArrayUtils.h"
 
 #define FORMAT_VERSION_0       0
 #define FORMAT_VERSION_1       1
@@ -96,37 +97,28 @@ struct DVBStreamInfo {
     LCID GetLCID() { return ISO6392ToLcid(CStringA(sLanguage)); };
 };
 
-static std::vector<CString> Supported_Protocols = [] {
-    std::vector<CString> Protocols;
-    Protocols.push_back(_T("rtp://"));
-    Protocols.push_back(_T("udp://"));
-    Protocols.push_back(_T("rtsp://"));
-    Protocols.push_back(_T("mms://"));
-    Protocols.push_back(_T("http://"));
-    Protocols.push_back(_T("https://"));
-    Protocols.push_back(_T("rtspu://"));
-    Protocols.push_back(_T("rtspm://"));
-    Protocols.push_back(_T("rtspt://"));
-    Protocols.push_back(_T("rtsph://"));
-    Protocols.push_back(_T("RTP://"));
-    Protocols.push_back(_T("UDP://"));
-    Protocols.push_back(_T("RTSP://"));
-    Protocols.push_back(_T("MMS://"));
-    Protocols.push_back(_T("HTTP://"));
-    Protocols.push_back(_T("HTTPS://"));
-    Protocols.push_back(_T("RTSPU://"));
-    Protocols.push_back(_T("RTSPM://"));
-    Protocols.push_back(_T("RTSPT://"));
-    Protocols.push_back(_T("RTSPH://"));
-    return Protocols;
-} ();
-
-
-static bool IsValidUrl(LPCTSTR strUrl)
+static bool IsValidUrl(CString strUrl)
 {
-    auto it = find_if(Supported_Protocols.begin(), Supported_Protocols.end(),
-        [&](CString const & Protocol) { return (((CString)strUrl).Find(Protocol) == 0); });
-    return (it != Supported_Protocols.end());
+    auto supportedProtocols = make_array(
+                                  _T("rtp://"),
+                                  _T("udp://"),
+                                  _T("rtsp://"),
+                                  _T("mms://"),
+                                  _T("http://"),
+                                  _T("https://"),
+                                  _T("rtspu://"),
+                                  _T("rtspm://"),
+                                  _T("rtspt://"),
+                                  _T("rtsph://"),
+                                  _T("rtmp://"),
+                                  _T("rtmpt://")
+                              );
+
+    auto it = find_if(supportedProtocols.cbegin(), supportedProtocols.cend(), [&](LPCTSTR protocol) {
+        return strUrl.MakeLower().Find(protocol) == 0;
+    });
+
+    return it != supportedProtocols.cend();
 }
 
 class CDVBChannel

```